sklearn/feature_extraction/text.py

```diff
@@ -685,9 +685,15 @@ def _sort_features(self, X, vocabulary):
         sorted_features = sorted(six.iteritems(vocabulary))
         map_index = np.empty(len(sorted_features), dtype=np.int32)
         for new_val, (term, old_val) in enumerate(sorted_features):
-            map_index[new_val] = old_val
             vocabulary[term] = new_val
-        return X[:, map_index]
+            map_index[old_val] = new_val
+
+        # swap columns in place
+        indices = X.indices
+        for idx, val in enumerate(X.indices):
+            indices[idx] = map_index[val]
+        X.indices = indices
+        return X
```

Comment:
@jnothman Thanks for the review! I agree with all your other comments, for this one though the documentation of  `CountVectorizer._sort_features` states, "Returns a reordered matrix and modifies the vocabulary in place", hence the `return X` (same as before the PR). Do you want me to change this behavior in the current PR ?